src/masternodes/rpc_icxorderbook.cpp

```diff
@@ -0,0 +1,1497 @@
+                            },
+                        },
+                    },
+                },
+                RPCResult{
+                        "\"hash\"                  (string) The hex-encoded hash of broadcasted transaction\n"
+                },
+                RPCExamples{
+                        HelpExampleCli("icx_claimdfchtlc", "'{\"dfchtlcTx\":\"<txid>>\",\"seed\":\"<seed>\"}'")
+                },
+     }.Check(request);
+
+    if (pwallet->chain().isInitialBlockDownload())
+        throw JSONRPCError(RPC_CLIENT_IN_INITIAL_DOWNLOAD, "Cannot claim dfc htlc while still in Initial Block Download");
+
+    pwallet->BlockUntilSyncedToCurrentChain();
+    LockedCoinsScopedGuard lcGuard(pwallet);
+
+    RPCTypeCheck(request.params, {UniValue::VOBJ}, false);
+    if (request.params[0].isNull())
+        throw JSONRPCError(RPC_INVALID_PARAMETER,
+                           "Invalid parameters, arguments 1 must be non-null and expected as object at least with "
+                           "{\"dfchtlcTx\",\"receiverAddress\",\"seed\"}");
+
+    UniValue metaObj = request.params[0].get_obj();
+    UniValue const & txInputs = request.params[1];
+
+    CICXClaimDFCHTLC claimdfchtlc;
+
+    if (!metaObj["dfchtlcTx"].isNull())
+        claimdfchtlc.dfchtlcTx = uint256S(metaObj["dfchtlcTx"].getValStr());
+    else
+        throw JSONRPCError(RPC_INVALID_PARAMETER,"Invalid parameters, argument \"dfchtlcTx\" must be non-null");
+    if (!metaObj["seed"].isNull())
+        claimdfchtlc.seed = ParseHex(metaObj["seed"].getValStr());
+    else
+        throw JSONRPCError(RPC_INVALID_PARAMETER,"Invalid parameters, argument \"seed\" must be non-null");
+
+    int targetHeight;
+    {
+        LOCK(cs_main);
+        auto dfchtlc = pcustomcsview->GetICXSubmitDFCHTLCByCreationTx(claimdfchtlc.dfchtlcTx);
+        if (!dfchtlc)
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "offerTx (" + claimdfchtlc.dfchtlcTx.GetHex() + ") does not exist");
+
+        std::vector<unsigned char> calcSeedBytes(32);
+        uint256 calcHash;
+        CSHA256()
+            .Write(claimdfchtlc.seed.data(),claimdfchtlc.seed.size())
+            .Finalize(calcSeedBytes.data());
+        calcHash.SetHex(HexStr(calcSeedBytes));
+
+        if (dfchtlc->hash != calcHash)
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "hash generated from given seed is different than in dfc htlc: " + calcHash.GetHex() + " - " + dfchtlc->hash.GetHex());
+        
+        targetHeight = ::ChainActive().Height() + 1;
+    }
+
+    CDataStream metadata(DfTxMarker, SER_NETWORK, PROTOCOL_VERSION);
+    metadata << static_cast<unsigned char>(CustomTxType::ICXClaimDFCHTLC)
+             << claimdfchtlc;
+
+    CScript scriptMeta;
+    scriptMeta << OP_RETURN << ToByteVector(metadata);
+
+    const auto txVersion = GetTransactionVersion(targetHeight);
+    CMutableTransaction rawTx(txVersion);
+
+    CTransactionRef optAuthTx;
+    std::set<CScript> auths;
```

Comment:
Do we need this an the subsequent auths.size() check? If not can be removed, auths in arg to GetAuthInputsSmart can be {} and auths.size() in conditional below can be removed.

Reply:
No, claim does not need auth as the other side is claiming htlc. Removed!